subworkflows/nf-scil/topup_eddy/tests/main.nf.test

```diff
@@ -0,0 +1,72 @@
+nextflow_workflow {
+
+    name "Test Subworkflow TOPUP_EDDY"
+    script "../main.nf"
+    workflow "TOPUP_EDDY"
+    config "./nextflow.config"
+
+    tag "subworkflows"
+    tag "subworkflows_nfcore"
+    tag "subworkflows/topup_eddy"
+
+    tag "preproc"
+    tag "preproc/topup"
+    tag "preproc/eddy"
+    tag "utils/extractb0"
+
+    tag "load_test_data"
+
+    setup {
+        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
+            script "../../load_test_data/main.nf"
+            process {
+                """
+                input[0] = Channel.from( [ "topup_eddy.zip" ] )
+                input[1] = "test.load-test-data"
+                """
+            }
+        }
+    }
+
+    test("topup_eddy") {
+
+        when {
+            workflow {
+                """
+                input[0] = LOAD_DATA.out.test_data_directory.map{
+                    test_data_directory -> [
+                        [ id:'test', single_end:false ],
+                        file("\${test_data_directory}/sub-01_dir-AP_dwi.nii.gz"),
+                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bval"),
+                        file("\${test_data_directory}/sub-01_dir-AP_dwi.bvec")
+                    ]}
+                input[1] = LOAD_DATA.out.test_data_directory.map{
+                    test_data_directory -> [
+                        [ id:'test', single_end:false ],
+                        file("\${test_data_directory}/sub-01_dir-AP_sbref.nii.gz")
+                    ]}
+                input[2] = LOAD_DATA.out.test_data_directory.map{
+                    test_data_directory -> [
+                        [ id:'test', single_end:false ],
+                        file("\${test_data_directory}/sub-01_dir-PA_dwi.nii.gz"),
+                        file("\${test_data_directory}/sub-01_dir-PA_dwi.bval"),
+                        file("\${test_data_directory}/sub-01_dir-PA_dwi.bvec")
+                    ]}
+                input[3] = LOAD_DATA.out.test_data_directory.map{
+                    test_data_directory -> [
+                        [ id:'test', single_end:false ],
+                        file("\${test_data_directory}/sub-01_dir-PA_sbref.nii.gz")
+                    ]}
+                input[4] = []
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success},
+                { assert snapshot(workflow.out).match()}
+            )
+        }
+    }
```

Comment:
It may be interesting to add another test without a reverse DWI or b0 to see the behavior of the subworkflow. If you want to add one, you don't need to redo a setup section for the test. Just add one here after the first `test ( "topup-eddy" )`.